src/com/zemoso/singleton/logger/Logger.java

```diff
@@ -0,0 +1,28 @@
+package com.zemoso.singleton.logger;
+
+public class Logger {
+
+    private static Logger loggerInstance;
+    private final int noOfPrints;
+
+    private Logger(int noOfPrints){
+        this.noOfPrints =noOfPrints;
+    }
+
+    public static Logger getLoggerInstance(int noOfPrints){
+        if (loggerInstance != null) {
```

Comment:
@nagavinay-zemoso please see if this works with multiple threads & does not create multiple instances, Thanks

Reply:
yeah I checked it with multiple threads, only single instance is getting created